src/main/webapp/app/overview/course-competencies/course-competencies-details.component.html

```diff
@@ -51,26 +51,28 @@ <h3 class="mb-0">
         <div class="row mt-2">
             <div class="col-lg-9 col-md-8">
                 <div class="row">
-                    @for (lectureUnit of competency.lectureUnits; track lectureUnit) {
-                        <div class="col-12 col-lg-11 mt-3">
-                            @switch (lectureUnit.type) {
-                                @case (LectureUnitType.EXERCISE) {
-                                    <jhi-exercise-unit [exerciseUnit]="lectureUnit" [course]="competency!.course!" />
+                    @for (lectureUnitLink of competency.lectureUnitLinks; track lectureUnitLink) {
+                        @if (lectureUnitLink.lectureUnit; as lectureUnit) {
+                            <div class="col-12 col-lg-11 mt-3">
+                                @switch (lectureUnit.type) {
+                                    @case (LectureUnitType.EXERCISE) {
+                                        <jhi-exercise-unit [exerciseUnit]="lectureUnit" [course]="competency!.course!" />
+                                    }
+                                    @case (LectureUnitType.ATTACHMENT) {
+                                        <jhi-attachment-unit [lectureUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)" />
+                                    }
+                                    @case (LectureUnitType.VIDEO) {
+                                        <jhi-video-unit [lectureUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)" />
+                                    }
+                                    @case (LectureUnitType.TEXT) {
+                                        <jhi-text-unit [lectureUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)" />
+                                    }
+                                    @case (LectureUnitType.ONLINE) {
+                                        <jhi-online-unit [lectureUnit]="lectureUnit" (onCompletion)="completeLectureUnit($event)" />
+                                    }
```

Comment:
Can we get rid of the errors here?

> Type LectureUnit is not assignable to type T 

If there is no other way to fix the type a directive similar to `exercise-detail.directive.ts` should solve the issue

Reply:
They were also present before this PR. We use some kind of polymorphism in a common super class that I do not want to touch in this PR